tests/golos.publication_rewards_types.hpp

```diff
@@ -343,7 +343,13 @@ struct limits {
                 return -1.0;
             vesting_ref -= price;
         }
-        return calc_power(limits::POSTBW, charges, cur_time, vesting_ref, 1.0);
+
+        if (kind != limits::COMM) {
```

Comment:
Why this strange condition?

Reply:
@s-medvedev, we don't need to change POSTBW for comments

Reply:
As I can see, `apply` function used in 2 places with `kind` equal COMM, POST and VOTE. This function has unclear logic (if say about code in general): it's difficult to understand what condition work in any case.
If say about this change: condition `kind != limits::COMM` assumes that this block will be calculate in all another cases. What about `VOTE`? We need return `power*power` in this case? Or this situation impossible and this condition never executes in this case? This code doesn't give answer for this question easily.

Reply:
There early exit for VOTE:
https://github.com/GolosChain/golos.contracts/blob/0803419d379f827b618029dc0956d96f67279842/tests/golos.publication_rewards_types.hpp#L334-L335

Agree, it's better to make this code clearer. Looks like it's also used if user pays for battery with vesting, and not sure it's tested…

Reply:
Create issue #725